practitioner-service/src/main/java/uk/ac/ox/ndph/arts/practitioner_service/service/PersonService.java

```diff
@@ -0,0 +1,48 @@
+package uk.ac.ox.ndph.arts.practitioner_service.service;
+
+import java.util.UUID;
+
+import org.hl7.fhir.r4.model.IdType;
+import org.hl7.fhir.r4.model.Practitioner;
+import org.hl7.fhir.r4.model.Enumerations.AdministrativeGender;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import uk.ac.ox.ndph.arts.practitioner_service.model.Person;
+import uk.ac.ox.ndph.arts.practitioner_service.repository.IFhirRepository;
+import uk.ac.ox.ndph.arts.practitioner_service.exception.HttpStatusException;
+import uk.ac.ox.ndph.arts.practitioner_service.exception.ArgumentException;
+
+/**
+* Implement an IEntityService.
+* validation is for empty values on all fields.
+*/
+@Service
+public class PersonService implements IEntityService {
+
+    private IFhirRepository fhirRepository;
+
+    @Autowired
+    public PersonService(IFhirRepository fhirRepository) {
+        this.fhirRepository = fhirRepository;
+    }
+
+    public String savePerson(Person person) throws HttpStatusException{
+        if (person.getPrefix() == null || person.getPrefix().isEmpty() || person.getGivenName() == null
+                || person.getGivenName().isEmpty() || person.getFamilyName() == null
+                || person.getFamilyName().isEmpty()) {
+            throw new ArgumentException("values cannot be null or empty");
```

Comment:
1. Should these be broken to return a more specific error?
2. I think you should do isBlank (and not is empty) to check whitespaces too.
3. Perhaps, create a method to do both null and blank checks or even go for a package line StringUtils (in Apache commons)

p.s. another approach could be to do this verification on the entity itself and not allow assignments of the instance properties if null or blank.

Reply:
regarding the p.s. comment. the Person DTO is the input to the API, it is generated by spring so I couldn't put validations there. additionally, the validation logic in US 178 should come here as well.